### PR TITLE
[12.0][add] account_bank_statement_import_txt_xlsx

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Import TXT/CSV and XLSX files as Bank Statements in Odoo

This module allows you to import any TXT, XLSX file in Odoo as bank
statements.

Configuration:
* Create or go to a bank journal where you want to import the txt statement.
* Edit that journal and set a Txt map in **Statement Import Txt Map** section in **Advanced
  Settings** tab.

* Now you can import Text based statements in that journal.

Note: if existent Txt Map does not fit to your file to import, you can
create another map in **Invoicing > Configuration > Accounting >
Statement Import Txt Mapping**.

You can import headers from any Txt file in **Action > Create Txt Map
Lines** and set every line with which field of statement have to match.

Usage:
To use this module, you need to:

#. Go to your bank online and download your Bank Statement in TXT/CSV or XLSX format.

This module has been created using the ideas from account_bank_statement_import_paypal

### CODE REVIEW

File: account_bank_statement_import_txt/tests/test_txt_statement_import.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        self.map = self.env['account.bank.statement.import.txt.map'].create({
+            'name': 'Txt Map Test',
+        })
+        usd = self.env.ref('base.USD')
+        self.journal = self.env['account.journal'].create({
+            'name': 'Txt Bank',
+            'type': 'bank',
+            'code': 'TXT',
+            'currency_id': (
+                usd.id if self.env.user.company_id.currency_id != usd else False
```
Comment (MiquelRForgeFlow) [inline account_bank_statement_import_txt/tests/test_txt_statement_import.py]: line too long

**Comment by JordiBForgeFlow** [comment]
@NuriaMartinXifre @joansis can you review?

**Comment by MiquelRForgeFlow** [comment]
Is this PR full ready? should something be squashed?

**Comment by MiquelRForgeFlow** [comment]
...

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by JoanSForgeFlow — APPROVED** [review]

**Pull request merged**